### ISSUE TITLE
boto3 service clients are not considered children of BaseClient used by Stubber

### ISSUE THREAD

**Issue by roberthl** [vemel/mypy_boto3_builder#92]
**Describe the bug**
boto3 service clients are not considered children of `BaseClient` type expected by `botocore.stub.Stubber.stub(client)`.

**To Reproduce**
1. Install 'boto3-stubs'
2.  Run mypy on the following code sample
```python
import boto3
from botocore.stub import Stubber

sns_client = boto3.client('sns')
stubber = Stubber(sns_client)
```
3. Actual mypy output

```
testcase.py:5: error: Argument 1 to "Stubber" has incompatible type "SNSClient"; expected "BaseClient"
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

```
Success: no issues found in 1 source file
```

**Additional context**

* Mac OS X
* Python 3.9.5
* boto3 1.17.102
* boto3-stubs 1.17.102.post1

**Comment by vemel** [comment]
Since botocore-stubs are also included and BaseClient is annotated, itake sense to use it as a base class now. There are some LSP violations, but not so many.

Thank you for the report, I will make changes it today.

**Comment by vemel** [comment]
Turns out, both LSP violations were because I did not generate `Client` class property. I have just released `mypy-boto3-builder 4.22.0` and new `boto3-stubs 1.17.103` should be generated in 3 hours or so.

Changes made:

- `Client.exceptions` is now a property
- `botocore.BaseClient.__getattr__` annotation removed because hides errors on non-existing method call
- `botocore.BaseClient.get_paginator` and `botocore.BaseClient.get_waiter` accert `Any` as an argument to be compatible with subclasses

**Comment by roberthl** [comment]
@vemel Thanks for getting to this so quickly.  I'm not seeing that release of boto3-stubs yet, but will try it out when it lands.

**Comment by vemel** [comment]
`boto3 1.17.103` released with corresponding `boto3-stubs 1.17.103`. Please update and let me know if your code works.

**Comment by roberthl** [comment]
Tested this out and working great now - cheers.
